plugins/beacon/ios/Sources/BaseBeaconPlugin.swift

```diff
@@ -94,23 +107,35 @@ open class BaseBeaconPlugin<BeaconStruct: Decodable>: JSBasePlugin {
         }
         let callback: @convention(block) (JSValue?) -> Void = { [weak self] rawBeacon in
             guard
-                let object = rawBeacon?.toObject(),
-                let data = try? JSONSerialization.data(withJSONObject: object),
-                let beacon = try? AnyTypeDecodingContext(rawData: data)
+                let object = rawBeacon?.toObject() else {
+                print("Failed to convert rawBeacon to object")
+                return
+            }
+            
+            print("Raw beacon object: \(object)")
+            
+            do {
+                let data = try JSONSerialization.data(withJSONObject: object)
+                print("Serialized data: \(String(data: data, encoding: .utf8) ?? "nil")")
+                
+                let beacon = try AnyTypeDecodingContext(rawData: data)
                     .inject(to: JSONDecoder())
                     .decode(BeaconStruct.self, from: data)
-            else { return }
-            self?.callback?(beacon)
+                
+                self?.callback?(beacon)
+            } catch {
+                print("Error during JSON serialization or deserialization: \(error)")
+            }
```

Comment:
This section should be reverted, we should not use `print` statements in code we publish

Reply:
Do you want me to remove all the print statements ? as I'm using some for debugging errors during the serialization process if any like catch {
                print("Error during JSON serialization or deserialization: \(error)")
            }

Reply:
fixed